### PR TITLE
[frr] Reduce Calls to SONiC Cfggen

### PR DESCRIPTION
**Pull request by tahmed-dev**
Calls to sonic-cfggen is CPU expensive. This PR reduces calls to
sonic-cfggen to two calls during startup when starting frr service.

singed-off-by: Tamer Ahmed <tamer.ahmed@microsoft.com>

**- Why I did it**
Reduce time required when invoking frr

**- How I did it**
Used template batch mode to batch together three calls into one call to sonic-cfggen

**- How to verify it**
```
root@str-s6000-acs-14:/# time ./docker_init_old.sh

real	0m12.873s
user	0m10.486s
sys	0m1.624s
root@str-s6000-acs-14:/# time ./docker_init_new.sh

real	0m3.572s
user	0m2.927s
sys	0m0.499s
root@str-s6000-acs-14:/# diff /etc/supervisor/conf.d/supervisord.conf.old /etc/supervisor/conf.d/supervisord.conf.new 
root@str-s6000-acs-14:/# diff /etc/frr/bgpd.conf.old /etc/frr/bgpd.conf.new 
root@str-s6000-acs-14:/# diff /etc/frr/zebra.conf.old /etc/frr/zebra.conf.new 
root@str-s6000-acs-14:/# diff /etc/frr/staticd.conf.old /etc/frr/staticd.conf.new 
root@str-s6000-acs-14:/# diff /usr/sbin/bgp-isolate.old /usr/sbin/bgp-isolate.new 
root@str-s6000-acs-14:/# diff /usr/sbin/bgp-unisolate.old /usr/sbin/bgp-unisolate.new 
root@str-s6000-acs-14:/# 

```

Update for the case when just once call is placed t sonic-cfggen:
```
root@str-s6000-acs-14:/# time ./docker_init_old.sh
CONFIG_TYPE=separated

real	0m12.117s
user	0m9.803s
sys	0m1.523s
root@str-s6000-acs-14:/# time ./docker_init_new.sh
CONFIG_TYPE=separated

real	0m2.598s
user	0m1.986s
sys	0m0.383s
root@str-s6000-acs-14:/# diff /etc/supervisor/conf.d/supervisord.conf.old /etc/supervisor/conf.d/supervisord.conf.new 
root@str-s6000-acs-14:/# diff /etc/frr/bgpd.conf.old /etc/frr/bgpd.conf.new 
root@str-s6000-acs-14:/# diff /etc/frr/zebra.conf.old /etc/frr/zebra.conf.new
root@str-s6000-acs-14:/# diff /etc/frr/staticd.conf.old /etc/frr/staticd.conf.new
root@str-s6000-acs-14:/# diff /usr/sbin/bgp-isolate.old /usr/sbin/bgp-isolate.new
root@str-s6000-acs-14:/# diff /usr/sbin/bgp-unisolate.old /usr/sbin/bgp-unisolate.new
root@str-s6000-acs-14:/# 
```

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [x] 201911
- [x] 202006

### CODE REVIEW

**Comment by tahmed-dev** [comment]
retest vsimage please

**Comment by tahmed-dev** [comment]
retest this please

**Comment by lguohan** [comment]
@pavel-shirshov  to review

File: dockers/docker-fpm-frr/docker_init.sh [hunk 1]
```diff
@@ -3,9 +3,19 @@
 mkdir -p /etc/frr
 mkdir -p /etc/supervisor/conf.d
 
-sonic-cfggen -d -t /usr/share/sonic/templates/supervisord/supervisord.conf.j2 > /etc/supervisor/conf.d/supervisord.conf
-
-CONFIG_TYPE=`sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]["docker_routing_config_mode"]'`
+CFGGEN_PARAMS=" \
+    -d \
+    -y /etc/sonic/constants.yml \
+    -t /usr/share/sonic/templates/supervisord/frr_vars.j2 \
```
Comment (pavel-shirshov) [inline dockers/docker-fpm-frr/docker_init.sh:9]: Why do we need frr_vars?
Also why you have only source after -t?

Comment (tahmed-dev) [reply dockers/docker-fpm-frr/docker_init.sh:9]: the frr_vars is a template file that pulls information about the config_type. This variable later decides if BGP service is integrate into vtysh or not. The content of the frr_vars is retreiving single variable for now:
`{{ DEVICE_METADATA["localhost"]["docker_routing_config_mode"] }}`
Lack of destination after the first argument, means this template is going to be rendered to stdout.

In comparison, commit [a5f2de](https://github.com/Azure/sonic-buildimage/pull/5176/commits/a5f2de02edb06bf2c8e34103753b1525258c672e) does docker initialization with two calls to cfggen. The time difference between the two approaches is about 1sec.

**Pull request merged**